x-pack/plugins/security_solution/public/resolver/test_utilities/simulator/index.tsx

```diff
@@ -146,6 +147,30 @@ export class Simulator {
       .forEach((wrapper) => wrapper.setState({ width: 10000, height: 10000 }));
   }
 
+  public requestNewTree(documentIdToFetch: string) {
+    this.store.dispatch({
+      type: 'appRequestedResolverData',
+      payload: documentIdToFetch,
+    });
+  }
+
+  public resolveNewTree(result: ResolverTree, fetchedDocumentID: string) {
+    this.store.dispatch({
+      type: 'serverReturnedResolverData',
+      payload: {
+        result,
+        databaseDocumentID: fetchedDocumentID,
+      },
+    });
+  }
+
+  public failToReturnTree(databaseDocumentID: string) {
```

Comment:
This is not working atm as `resolverTreeFetcher` in the middleware requests the initial existing data when the simulator was initialized, so while this triggers a failure it is then overriden